openpgp/internal/ecc/curve25519.go

```diff
@@ -25,30 +23,66 @@ func (c *curve25519) GetCurveName() string {
 	return "curve25519"
 }
 
-func (c *curve25519) MarshalPoint(x, y *big.Int) []byte {
-	return x.Bytes()
+// MarshalBytePoint encodes the public point from native format, adding the prefix.
+// See https://datatracker.ietf.org/doc/html/draft-ietf-openpgp-crypto-refresh-06#section-5.5.5.6
+func (c *curve25519) MarshalBytePoint(point [] byte) []byte {
+	return append([]byte{0x40}, point...)
 }
 
-func (c *curve25519) UnmarshalPoint(point []byte) (x, y *big.Int) {
+// UnmarshalBytePoint decodes the public point to native format, removing the prefix.
+// See https://datatracker.ietf.org/doc/html/draft-ietf-openpgp-crypto-refresh-06#section-5.5.5.6
+func (c *curve25519) UnmarshalBytePoint(point []byte) []byte {
 	if len(point) != x25519lib.Size + 1 {
-		return nil, nil
+		return nil
 	}
 
-	x = new(big.Int)
-	x.SetBytes(point)
-	return x, nil
+	// Remove prefix
+	return point[1:]
 }
 
-func (c *curve25519) MarshalByteSecret(d []byte) []byte {
+// MarshalByteSecret encodes the secret scalar from native format.
+// Note that the EC secret scalar differs from the definition of public keys in
+// [Curve25519] in two ways: (1) the byte-ordering is big-endian, which is
+// more uniform with how big integers are represented in TLS, and (2) the
+// leading zeros are truncated.
+// See https://datatracker.ietf.org/doc/html/draft-ietf-openpgp-crypto-refresh-06#section-5.5.5.6.1.1
+// Here Leading zero bytes are stripped when encoding to MPI.
```

Comment:
This comment is also a bit confusing imo because the leading zeros are not stripped here, but later (when encoded as an MPI as you say). So I would say something like
```suggestion
// Note that leading zero bytes are stripped later when encoding as an MPI.
```